power_grid_model_c/power_grid_model/include/power_grid_model/container.hpp

```diff
@@ -272,6 +292,15 @@ class Container<RetrievableTypes<GettableTypes...>, StorageableTypes...> {
         return res;
     }
 
+    template <class Storageable>
+    void restore_values_impl() {
+        auto& cached_vec = std::get<std::vector<std::pair<Idx, Storageable>>>(cached_reset_values_);
+        for (auto const& cache : cached_vec) {
```

Comment:
As commented above, reverse restore the object garantees that the original object is restored, if (in rare case) one object is updated multiple times.

```suggestion
        for (auto it = cached_vec.rcbegin(); it != cached_vec.rcend(); ++it) {
            auto const& cache = *it;
```